my_sprites.py

```diff
@@ -858,3 +869,71 @@ def update(self):
 
             # Time passes
             self._time_to_idle -= 1/60  # we don't want to use on_update, so we just use the default delta time
+
+
+class HealthBar(arcade.Sprite):
+
+    def __init__(self,  max_health, center_x=0, center_y=0, bar_width=32, bar_height=5, offset=15, scale=1):
+
+        super().__init__(
+            center_x=center_x,
+            center_y=center_y,
+            scale=scale,
+        )
+
+        # variable controlling length of fullness of health bar
+        self._max_health = max_health
+        self._current_health = self._max_health
+
+        self._bar_width = bar_width * scale
+        self._bar_height = bar_height * scale
+
+        self._offset = offset # y offset, to offset the bar, so it is not drawn on top of the player
+
+
+        # static bar behind the dynamic bar
+
+        self._background_bar = arcade.SpriteSolidColor(
+            self._bar_width,
+            self._bar_height,
+            arcade.color.RED
+        )
+
+        # bar changing depending on the percentage variable
+
+        self._foreground_bar = arcade.SpriteSolidColor(
+            self._bar_width,
+            self._bar_height,
+            arcade.color.GREEN
+        )
+
+    @property
+    def max_health(self):
+        return self._max_health
+
+    @property
+    def health(self):
+        return self._current_health
+
+    @health.setter
+    def health(self, new_health):
```

Comment:
This feels wrong. If max_health is 100 and I set health to  50 the value of _current_health will be set to 50/100 = 1/2.
So if I set _bar.health = 50_  and then read _bar.health_ that value will now be _1/2_, not the 50 that I just set it to?

The _bar.health_ should be your _hp_, not some ratio.